Drivers/Src/Display.c

```diff
@@ -1,325 +1,280 @@
+	return Display_SetComponent(ARRAY, (uint8_t)state);
 }
 
-ErrorStatus Display_SetGear(State fwd, State rev){
-    if(!fwd && !rev){
-        setComponentVisibility(NEUTRAL_BOX, true);
-        setComponentVisibility(FWD_BOX, false);
-        return setComponentVisibility(REV_BOX, false);
-    }
-    else if(fwd && !rev){
-        setComponentVisibility(NEUTRAL_BOX, false);
-        setComponentVisibility(FWD_BOX, true);
-        return setComponentVisibility(REV_BOX, false);
-    }
-    else if(rev && !fwd){
-        setComponentVisibility(NEUTRAL_BOX, false);
-        setComponentVisibility(REV_BOX, true);
-        return setComponentVisibility(FWD_BOX, false);
-    }
-    else{
-        return ERROR;
-    }
+ErrorStatus Display_SetMotor(bool state){
+	return Display_SetComponent(MOTOR, (uint8_t)state);
 }
 
+ErrorStatus Display_SetRegenEnable(bool state){
+	return Display_SetComponent(MOTOR, (uint8_t)state);
+}
 
-/**
- * Set ERROR<idx> to err
- * If err is an empty string, then the error will be cleared
- */
-ErrorStatus Display_SetError(int idx, char *err)
-{
-    if (idx < 0 || idx > 5)
-        return ERROR; // Index out of bounds
-    if (strlen(err) == 0)
-    {
-        CHECK(setComponentVisibility(ERROR0 + idx, false)); // Hide text if no error
-    }
-    else
-    {
-        CHECK(updateStringValue(ERROR0 + idx, TEXT, err)); // Set error string
-        CHECK(updateIntValue(ERROR0 + idx, PCO, NEXTION_RED));
-        CHECK(setComponentVisibility(ERROR0 + idx, true));
-    }
-    return SUCCESS;
+ErrorStatus Display_SetCruiseEnable(bool state){
+	return Display_SetComponent(CRUISE, (uint8_t)state);
 }
 
-/**
- * Set the first error slot to say "No Error" and set its color to green
- * User must clear the remaining slots manually using Display_SetError
- */
-ErrorStatus Display_NoErrors(void)
-{
-    CHECK(updateStringValue(ERROR0, TEXT, (char *)NO_ERROR));
-    CHECK(updateIntValue(ERROR0, PCO, NEXTION_GREEN));
-    CHECK(setComponentVisibility(ERROR0, true));
-    return SUCCESS;
+ErrorStatus Display_SetLeftBlink(bool state){
+	return Display_SetComponent(LEFT, (uint8_t)state);
 }
 
-/**
- * Set the display to the main view
- */
-ErrorStatus Display_SetMainView(void)
-{
-    //return updateIntValue(SYSTEM, PAGE, 1);
-    char *page = "page1\xff\xff\xff";
-    BSP_UART_Write(DISP_OUT, page, strlen(page));
-    return SUCCESS;
+ErrorStatus Display_SetRightBlink(bool state){
+	return Display_SetComponent(RIGHT, (uint8_t)state);
 }
 
-ErrorStatus Display_SetLight(uint8_t light, State on){
-    if(light == 2){  // ctrl fault
-        if(on==ON)
-            return updateIntValue(PAGE, BCO, NEXTION_RED);
-        return updateIntValue(PAGE, BCO, NEXTION_DARK_GREY);
-    }
-        
-    setComponentVisibility(light+19, on==ON);
-    return SUCCESS;
+ErrorStatus Display_Fault(os_error_loc_t osErrCode, fault_bitmap_t faultCode){
+	char faultPage[7] = "page 2";
+	BSP_UART_Write(DISP_OUT, faultPage, strlen(faultPage));
```

Comment:
This might fail if the display is in the middle of receiving a command. It might be a good idea to send a terminator on its own to end the (potential) previous command and put the display in a consistent state we can talk to.